code/modules/projectiles/guns/projectile/automatic.dm

```diff
@@ -26,20 +27,13 @@
 	if(istype(A, /obj/item/ammo_box/magazine))
 		var/obj/item/ammo_box/magazine/AM = A
 		if(istype(AM, mag_type))
-			if(magazine)
+			if(!can_reload())
 				to_chat(user, "<span class='notice'>You perform a tactical reload on \the [src], replacing the magazine.</span>")
 				magazine.loc = get_turf(loc)
 				magazine.update_icon()
-				magazine = null
-			else
-				to_chat(user, "<span class='notice'>You insert the magazine into \the [src].</span>")
-			user.remove_from_mob(AM)
-			magazine = AM
-			magazine.loc = src
-			chamber_round()
-			A.update_icon()
-			update_icon()
-			return 1
+				magazine = null		
+			reload(AM, user)			
+			return

```

Comment:
This part can be moved to the attackby from projectile due to the reason stated before.